compiler/rustc_const_eval/src/const_eval/error.rs

```diff
@@ -34,7 +34,17 @@ impl MachineStopType for ConstEvalErrKind {
         match self {
             ConstAccessesMutGlobal => const_eval_const_accesses_mut_global,
             ModifiedGlobal => const_eval_modified_global,
-            Panic { .. } => const_eval_panic,
+            Panic { msg, .. } => {
+                let msg = msg.as_str();
+                match msg {
+                    "explicit panic"
+                    | "not implemented"
+                    | "not yet implemented"
+                    | "internal error: entered unreachable code" => msg.to_string().into(),
+                    _ if msg.starts_with("assertion failed: ") => msg.to_string().into(),
+                    _ => const_eval_panic,
```

Comment:
Uh, what a hack... is this really needed?

Reply:
Is it absolutely needed? No. But I have distaste of the "panic: explicit panic" output, *and* the was a prior request for adding the "panic: " prefix to all panics. I would personally prefer dropping "panic: " in all cases over the "panic: explicit panic" output.

Reply:
But we can't just hard-code some random strings here. That goes against all decent engineering practices. I don't want to have a hack like that anywhere near code I am responsible for.

A runtime panic looks like this:
```
thread 'main' panicked at src/main.rs:2:5:
explicit panic
```
Note that is also says "panic" twice. So I see no good reason to try and do something else in const-eval.

Reply:
We should probably emit `evaluation panicked: $msg` or so, that seems most consistent with the runtime wording.